mongoengine/document.py

```diff
@@ -300,11 +288,13 @@ def save(self, force_insert=False, validate=True, clean=True,
             collection = self._get_collection()
             if self._meta.get('auto_create_index', True):
                 self.ensure_indexes()
-            if created:
-                if force_insert:
-                    object_id = collection.insert(doc, **write_concern)
-                else:
-                    object_id = collection.save(doc, **write_concern)
+            if '_id' not in doc:
+                object_id = collection.insert_one(doc).inserted_id
+            elif created:
+                # If created is true then the delta will not be in sync with the
+                # document; we must upsert the document to preserve all fields.
```

Comment:
@mklein0: please let me know if the new version of the comment makes more sense.